diff --git a/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js b/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js
index df60738..c392de6 100644
--- a/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js
+++ b/node_modules/@expo/cli/build/src/export/embed/exportEmbedAsync.js
@@ -181,7 +181,7 @@ async function exportEmbedBundleAndAssetsAsync(projectRoot, options) {
             // source map inline
             reactCompiler: !!((ref = exp.experiments) == null ? void 0 : ref.reactCompiler)
         }, {
-            sourceMapUrl,
+            // sourceMapUrl,
             unstable_transformProfile: options.unstableTransformProfile || (isHermes ? "hermes-stable" : "default")
         });
         return {
