diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverter.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverter.kt
index a17efa5..d3d06b2 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverter.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverter.kt
@@ -29,6 +29,7 @@ object JSTypeConverter {
       is Bundle -> value.toJSValue(containerProvider)
       is Array<*> -> value.toJSValue(containerProvider)
       is IntArray -> value.toJSValue(containerProvider)
+      is LongArray -> value.toJSValue(containerProvider)
       is FloatArray -> value.toJSValue(containerProvider)
       is DoubleArray -> value.toJSValue(containerProvider)
       is BooleanArray -> value.toJSValue(containerProvider)
diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverterHelper.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverterHelper.kt
index 9e9be54..03df1ff 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverterHelper.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverterHelper.kt
@@ -90,6 +90,14 @@ fun IntArray.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): Wr
   }
 }
 
+fun LongArray.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableArray {
+  return containerProvider.createArray().also {
+    for (value in this) {
+      it.pushInt(value.toInt())
+    }
+  }
+}
+
 fun FloatArray.toJSValue(containerProvider: JSTypeConverter.ContainerProvider): WritableArray {
   return containerProvider.createArray().also {
     for (value in this) {
