diff --git a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java
index 3ae6481..a03e70d 100644
--- a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java
+++ b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java
@@ -11,6 +11,8 @@ import android.util.Log;
 
 import androidx.core.app.NotificationCompat;
 
+import java.util.Objects;
+
 import expo.modules.core.interfaces.ReactActivityLifecycleListener;
 import expo.modules.notifications.notifications.NotificationManager;
 import expo.modules.notifications.notifications.model.Notification;
@@ -70,7 +72,10 @@ public class ExpoNotificationLifecycleListener implements ReactActivityLifecycle
         Log.d("ExpoNotificationLifecycleListener", method + " : keys count = " + extra.keySet().size());
 
         for (String key : extra.keySet()) {
-            Log.d("ExpoNotificationLifecycleListener", method + " : key = " + key + " = " + extra.get(key).toString());
+            Log.d(
+                    "ExpoNotificationLifecycleListener",
+                    method + " : key = " + key + " = " + Objects.requireNonNull(extra.get(key)).toString()
+            );
         }
     }
 }
diff --git a/node_modules/expo-notifications/plugin/build/withNotificationsAndroid.js b/node_modules/expo-notifications/plugin/build/withNotificationsAndroid.js
index 9ca09dc..45bc261 100644
--- a/node_modules/expo-notifications/plugin/build/withNotificationsAndroid.js
+++ b/node_modules/expo-notifications/plugin/build/withNotificationsAndroid.js
@@ -17,8 +17,8 @@ exports.dpiValues = {
 const { addMetaDataItemToMainApplication, getMainApplicationOrThrow, removeMetaDataItemFromMainApplication, } = config_plugins_1.AndroidConfig.Manifest;
 const BASELINE_PIXEL_SIZE = 24;
 const ERROR_MSG_PREFIX = 'An error occurred while configuring Android notifications. ';
-exports.META_DATA_NOTIFICATION_ICON = 'expo.modules.notifications.default_notification_icon';
-exports.META_DATA_NOTIFICATION_ICON_COLOR = 'expo.modules.notifications.default_notification_color';
+exports.META_DATA_NOTIFICATION_ICON = 'com.google.firebase.messaging.default_notification_icon';
+exports.META_DATA_NOTIFICATION_ICON_COLOR = 'com.google.firebase.messaging.default_notification_color';
 exports.NOTIFICATION_ICON = 'notification_icon';
 exports.NOTIFICATION_ICON_RESOURCE = `@drawable/${exports.NOTIFICATION_ICON}`;
 exports.NOTIFICATION_ICON_COLOR = 'notification_icon_color';
diff --git a/node_modules/expo-notifications/plugin/src/withNotificationsAndroid.ts b/node_modules/expo-notifications/plugin/src/withNotificationsAndroid.ts
index 94d2bd0..581e165 100644
--- a/node_modules/expo-notifications/plugin/src/withNotificationsAndroid.ts
+++ b/node_modules/expo-notifications/plugin/src/withNotificationsAndroid.ts
@@ -32,9 +32,10 @@ const {
 } = AndroidConfig.Manifest;
 const BASELINE_PIXEL_SIZE = 24;
 const ERROR_MSG_PREFIX = 'An error occurred while configuring Android notifications. ';
-export const META_DATA_NOTIFICATION_ICON = 'expo.modules.notifications.default_notification_icon';
+export const META_DATA_NOTIFICATION_ICON =
+  'com.google.firebase.messaging.default_notification_icon';
 export const META_DATA_NOTIFICATION_ICON_COLOR =
-  'expo.modules.notifications.default_notification_color';
+  'com.google.firebase.messaging.default_notification_color';
 export const NOTIFICATION_ICON = 'notification_icon';
 export const NOTIFICATION_ICON_RESOURCE = `@drawable/${NOTIFICATION_ICON}`;
 export const NOTIFICATION_ICON_COLOR = 'notification_icon_color';
